### PR TITLE
feat(cpu type): CPU type support for EPYC-v4

### PR DESCRIPTION
**Pull request by Mrton0121**
### Contributor's Note

- [x] I have added / updated documentation in `/docs` for any user-facing features or additions.
- [ ] I have added / updated acceptance tests in `/fwprovider/tests` for any new or updated resources / data sources.
- [x] I have ran `make example` to verify that the change works as expected.

### Community Note

- Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #0000 | Relates #0000

### CODE REVIEW

**Comment by bpg** [comment]
Hi @Mrton0121 👋🏼 
Thanks for adding this! Would you mind signing the commits for [DCO](https://github.com/bpg/terraform-provider-proxmox/pull/1611/checks?check_run_id=32337787295), so I can merge the PR?

File: go.mod
Comment (bpg-dev) [inline go.mod:1]: this seems to be an accidental change updating the file mode to executable, could you revert it?

Comment (Mrton0121) [reply go.mod:1]: Hi! Reverted the change on go.mod

**Review by bpg — APPROVED** [review]
Thank you!

LGTM! 🚀

**Comment by bpg** [comment]
@all-contributors please add @Mrton0121 for code

**Pull request merged**